### PR TITLE
Fix #2340:  Ingestion pipeline deployment refactor

### LINKED ISSUE #2340: Ingestion pipeline deployment refactor
1. Remove JDBC from Services
2. Add DatabaseConnection to Service. Right now this is one per service
3. Users can add Advanced configs through ConnectionOptions and ConnectionArguments
4. A Service can contain multiple pipelines that can be deployed to Airflow
5. Right now we are going to support Metadata Extraction and Usage pipelines and in next release we will add Profiler as separate ingestion job and data quality pipelines as well
6. ingestion_entity is renamed as airflow_pipeline_entity

### PR DESCRIPTION
**Pull request by harshach**
### Describe your changes :
See #2340 

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: catalog-rest-service/src/main/resources/json/schema/operations/pipelines/airflowPipeline.json [hunk 1]
```diff
@@ -0,0 +1,189 @@
+{
+  "$id": "https://open-metadata.org/schema/operations/pipelines/airflowPipeline.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Airflow Pipeline",
+  "description": "Airflow Pipeline is used to setup a DAG and deploy.",
```
Comment (mithmatt) [inline catalog-rest-service/src/main/resources/json/schema/operations/pipelines/airflowPipeline.json:5]: This entity will be only used for ingestion and operations required for OpenMetadata

And the other schema/data/entity/pipeline.json will be used for the typical pipeline assets as usual (even if they are airflow workflows that do ETL).

If yes, should we mention this in schema description to be explicit (although it is nested under operations)?

Comment (harshach) [reply catalog-rest-service/src/main/resources/json/schema/operations/pipelines/airflowPipeline.json]: done

**Comment by akash-jain-10** [comment]
> @akash-jain-10 @pmbrull do we need twisted lib dependency on incrementing a simple version? python -m incremental.update metadata --dev Traceback (most recent call last): File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 197, in _run_module_as_main return _run_code(code, main_globals, None, File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 87, in _run_code exec(code, run_globals) File "/Users/harsha/Code/OpenMetadata/ingestion/env/lib/python3.9/site-packages/incremental/update.py", line 11, in from twisted.python.filepath import FilePath ModuleNotFoundError: No module named 'twisted'

Yes. Twisted is a dependency for python package incremental. We will require this as dependency

**Review by ayush-shah — APPROVED** [review]

**Comment by pmbrull** [comment]
> > @akash-jain-10 @pmbrull do we need twisted lib dependency on incrementing a simple version? python -m incremental.update metadata --dev Traceback (most recent call last): File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 197, in _run_module_as_main return _run_code(code, main_globals, None, File "/usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 87, in _run_code exec(code, run_globals) File "/Users/harsha/Code/OpenMetadata/ingestion/env/lib/python3.9/site-packages/incremental/update.py", line 11, in from twisted.python.filepath import FilePath ModuleNotFoundError: No module named 'twisted'
> 
> Yes. Twisted is a dependency for python package incremental. We will require this as dependency

It is present in the setup for ingestion core. We have some instructions in the core README but we could add further information on the bot message to line up all the steps or even work further on the make recipe to run the installation etc. I'll open an issue for this. Thanks!

**Pull request merged**